### ISSUE TITLE
Errors are displayed in JSX2->JSX3 migration projects, even when project compiles fine

### ISSUE THREAD

**Issue by briangorman** [jaredly/reason-language-server#301]
I have created an example project to demonstrate this:

https://github.com/briangorman/reason-lsp-demo

The rough steps to reproduce:
1. Create JSX2 component
2. Create JSX3 child component 
3. Create JSX3 child component for component in step 2
4. Observe how reason-language server reports an error in the initial JSX3 child component.

https://github.com/briangorman/reason-lsp-demo

Some error like this will be displayed:

```
This function has type {. "id": string} => ReasonReact.reactElement
       It is applied to too many arguments; maybe you forgot a `;'

```

However the project compiles/runs fine.

**Comment by alexfedoseev** [comment]
I think I'm facing related issue. `[@bs.config]` attribute is ignored by rls: https://cl.ly/a983ab039135
`bsb` compiles project without warnings.

**Comment by mrmurphy** [comment]
We just ran into the same issue.

**Comment by jaredly** [comment]
😕 I think this is probably a bucklescript issue, but I'm not 100% sure. I would assume that the `bsppx.exe` ppx that bucklescript bundles is supposed to see the `[@bs.config]` pragma and do something different

**Comment by jaredly** [comment]
Ok digging into it more, and I'm not sure there's a way for me to fix this on my end 😕
here's what I've found:
- bsc [recently added](https://github.com/BuckleScript/bucklescript/commit/071bad06b4ef6664645b312ef3cdc648d311b815) a `-bs-jsx` flag so that it can natively handle jsx, which is something that previously the `bsppx` binary took care of
- as part of this (I assume), bsb is producing a `ninja` build file that no longer references `bsppx`
- however, it's still generating a `.merlin` file that indicates that `bsppx` should be run on all files
- if I try to have rls manually ignore the `bsppx` flag, then other things start breaking (`##`, `->`, and other things that bsppx is responsible for).

cc @bobzhang @chenglou little help?

**Comment by mrmurphy** [comment]
I know this probably isn't an option for really huge projects, but we ended up migrating all of our components to JSX 3 yesterday instead of dealing with this issue. At a kind recommendation from a chat user, we followed this approach and it took just a few minutes to complete the migration:

https://gist.github.com/mrmurphy/8cde6163c301acfd92d60198eb7fcaef

**Comment by bobzhang** [comment]
>  I would assume that the bsppx.exe ppx that bucklescript bundles is supposed to see the [@bs.config] pragma and do something different

If this is the case, it should be fixed, can you confirm that?

Btw, I suggest that bsb should always be the source of true for error reporting, while rls is more valuable in auto-completion etc

**Comment by bobzhang** [comment]
It seems the way of reason-jsx retriving jsx version is not correct under ppx flag, (cc @rickyvetter)
The ocaml compiler will inject some attributes, see below
```
[@@@ocaml.ppx.context
  {
    tool_name = "BuckleScript";
    include_dirs =
      ["/Users/hongbozhang/git/reason-lsp-demo/node_modules/bs-platform/lib/ocaml"];
    load_path =
      ["/Users/hongbozhang/git/reason-lsp-demo/node_modules/bs-platform/lib/ocaml"];
    open_modules = [];
    for_package = None;
    debug = true;
    cookies = []
  }]
[@@@bs.config { jsx = 3 }]
```

**Comment by bobzhang** [comment]
This is fixed in master and will be released in 5.0.6. 
The suggestion is still recommeded:

>Btw, I suggest that bsb should always be the source of true for error reporting, while rls is more valuable in auto-completion etc.
